aggregator/src/aggregator/query_type.rs

```diff
@@ -240,12 +247,26 @@ pub trait CollectableQueryType: AccumulableQueryType {
         )?;
         Ok(batch_aggregations.into_iter().flatten().collect::<Vec<_>>())
     }
+
+    async fn acknowledge_collection<C: Clock>(
```

Comment:
I think this needs a doccomment. In particular, to explain when in the collection state machine this call would occur. If I was just reading this trait, I might think that this method is an appropriate place to consume max query count, but that needs to happen during the computation of the aggregate share, not here.